### DISCUSSION TITLE
Are the text-opacity classes deprecated?

### DISCUSSION THREAD

**Discussion by chasegiunta** [tailwindlabs/tailwindcss#7512] [General]
I noticed the `text-opacity-###` classes are no longer mentioned in the Tailwind 3 docs. Just curious whether this was intentionally left out, an oversight, or just didn't get to it perhaps? I _do_ find the text opacity classes helpful in cases where the context (say within a component) isn't afforded the current color, or simply not having to redeclare the color for the sake of cleanliness.

**Comment by adamwathan** [comment]
Kinda undecided, we stopped documenting just to sort of see how much we'd hear about it and find out about situations where it's really necessary. No actual plans to remove them I don't think, but if they didn't already exist we probably wouldn't reintroduce them. CSS variables are sort of expensive at run-time if you get really carried away with them (especially in Safari) so it would be nice to be able to give something a background color without adding a variable just in case someone wants to set the opacity.

**Comment by augustl** [comment]
Seems like this has now been removed in Tailwind v4, and the migration tool didn't pick it up. I have an `@apply disabled:text-opacity-50` in a .css file, and it's causing this error in my next.js app:

```
Error evaluating Node.js code
Error: Cannot apply unknown utility class: disabled:text-opacity-50
```

It's easily resolvable, I just replaced `text-black disabled:text-opacity-50` with `text-black disabled:text-black/50` etc.
